File: 45. Jump Game II/Solution.java
class Solution {
    public int jump(int[] nums) {
        int n = nums.length, curr = 0, high = 0, count = 0;
        if (n == 1) return 0;
        while (high < n - 1) {
            int nextHigh = high, nextCurr = curr;
            for (int i = curr; i <= high; i++) {
                if (i + nums[i] > nextHigh) {
                    nextHigh = i + nums[i];
                    nextCurr = i;
                }
            }
            curr = nextCurr;
            high = nextHigh;
            count++;
        }
        return count;
    }
}

File: 45. Jump Game II/Solution.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n, curr, high, count = len(nums), 0, 0, 0
        while high < n - 1:
            nextCurr, nextHigh = curr, high
            for i in range(curr, high + 1):
                if i + nums[i] > nextHigh:
                    nextHigh, nextCurr = i + nums[i], i
            high, curr = nextHigh, nextCurr
            count += 1
        return count